shell/platform/windows/windows_lifecycle_manager.cc

```diff
@@ -163,4 +181,65 @@ void WindowsLifecycleManager::BeginProcessingClose() {
   process_close_ = true;
 }
 
+void WindowsLifecycleManager::SetLifecycleState(AppLifecycleState state) {
+  if (state_ == state) {
+    return;
+  }
+  state_ = state;
+  if (engine_) {
+    const char* state_name = AppLifecycleStateToString(state);
+    engine_->SendPlatformMessage("flutter/lifecycle",
+                                 reinterpret_cast<const uint8_t*>(state_name),
+                                 strlen(state_name), nullptr, nullptr);
+  }
+}
+
+void WindowsLifecycleManager::OnWindowStateEvent(HWND hwnd,
+                                                 WindowStateEvent event) {
+  // Synthesize an unfocus event when a focused window is hidden.
+  if (event == WindowStateEvent::kHide &&
+      focused_windows_.find(hwnd) != focused_windows_.end()) {
+    OnWindowStateEvent(hwnd, WindowStateEvent::kUnfocus);
+  }
+
+  std::lock_guard guard(state_update_lock_);
+  switch (event) {
+    case WindowStateEvent::kShow: {
+      bool first_shown_window = visible_windows_.empty();
+      auto pair = visible_windows_.insert(hwnd);
+      if (first_shown_window && pair.second &&
+          state_ == AppLifecycleState::kHidden) {
+        SetLifecycleState(AppLifecycleState::kInactive);
```

Comment:
Is this the right event?

Reply:
I think so. In other words, if a window has just been shown, and there were previously no visible windows, and we are currently in the hidden state, enter the inactive state. Does this sound incorrect or not represented by the above code?

Reply:
Is the app focused if I click on the app from the Windows taskbar? In what scenarios would an app be shown but not focused?

Since it's not readily apparent why the app receives the `Inactive` state when a window is shown, I would add a quick comment that the resumed state is sent when the view's `HWND` is focused.

Reply:
Under most circumstances there may not be a scenario when a window goes from hidden/minimized to shown/restored without gaining focus, but it is always possible for a window to be explicitly shown through `ShowWindow` or restored programmatically, which does not necessarily affect focus. Also, I believe @gspencergoog mentioned at one point it is preferable to send the framework a notification of entering the inactive state in between transitioning from hidden to resumed.

Reply:
The framework will automatically generate a transition to inactive in between if you go from hidden to resumed, but I'd prefer that the embedding generate the transition itself in response to real events.

The idea is that all platforms have the same state machine, so application writers can depend on it.

Reply:
This seems reasonable, marking as resolved